### PR TITLE
Refs #24121 -- Added __repr__() to Message.

### PR DESCRIPTION
**Pull request by francoisfreitag**

### CODE REVIEW

File: django/contrib/messages/storage/base.py [hunk 1]
```diff
@@ -32,6 +32,11 @@ def __eq__(self, other):
     def __str__(self):
         return str(self.message)
 
+    def __repr__(self):
+        if self.level_tag:
+            return f"[{self.level_tag}] {self.message}"
+        return self.message
```
Comment (felixxm) [inline django/contrib/messages/storage/base.py:38]: According to [Python docs](https://docs.python.org/3/reference/datamodel.html#object.__repr__):

_"If at all possible, this should look like a valid Python expression that could be used to recreate an object with the same value (given an appropriate environment)"_

Maybe:
```python
extra_tags = ""
if self.extra_tags:
    extra_tags =f ", extra_tags={self.extra_tags}"
return f"Message({self.level}, {self.message}{extra_tags})
```

Comment (shangxiao) [reply django/contrib/messages/storage/base.py:38]: Just looking at various `__repr__`s throughout the codbase, looks like there's some inconsistency; some use an angled brackets notation, some using new object notation 😊

I think felix's suggestion is best 👍

From the existing examples I saw use keyword notation:

`Message(level={level}, message={message}{extra_tags})`

Comment (francoisfreitag) [reply django/contrib/messages/storage/base.py]: Thanks for pointing that out!
I added quotes, the same way `decimal.Decimal` behaves for instance. If the goal is to be able to copy/paste into a REPL to reconstruct the object, having the quotes make things easier, and also helps better delimiting fields.

Comment (felixxm) [reply django/contrib/messages/storage/base.py]: Agreed, changed to `!r`.

**Comment by felixxm** [comment]
@francoisfreitag Thanks :+1:

**Comment by felixxm** [comment]
Merged in 7bba231b67996fce294f257e1316748cc4ef299a.

**Pull request closed without merging**